File: AdministracionDeTienda/src/gestorAplicacion/clasesBase/Producto.java
package gestorAplicacion.clasesBase;

import java.io.Serializable;
import java.util.ArrayList;

import gestorAplicacion.clasesBase.Bodega.SETS;


public class Producto implements Serializable{

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	private String nombre;
	private double precio;
	private double costo;
	private Tipo tipo;

	
	
	//private double costo;


	
	
	public Producto(Tipo tipo) {
		this.nombre=tipo.toString();
		this.precio=tipo.getPrecioEstandar();
		//Constructor para socio
	}
	
	public Producto(Tipo tipo, double precio, double costo) {
		
		this(tipo);
		this.precio=precio;
		this.costo = costo;
		this.tipo=tipo;
		
		
	}

	

	public Producto(Tipo tipo, double costo) {
		this.tipo=tipo;
		this.costo=costo;
		//Constructor para intervenido
	}

	public boolean equals(Producto p) {
		
		if(p.getNombre()==this.nombre) {

			return true;
		}
		else {
			return false;
		}
		
	}
	
	public void descontar( float dcto) {
		if(dcto<0.7 || dcto >=1) {
			this.setPrecio(getPrecio()*0.9);
		}
		else {
		this.setPrecio(getPrecio()*dcto);
		}
	}
	
	
	
	public double getPrecio() {
		return precio;
	}




	public void setPrecio(double precio) {
		this.precio = precio;
	}




	public String getNombre() {
		return this.nombre;
	}
	
	public void setNombre(String nombre) {
		this.nombre=nombre;
		
	}
	
	public void setCosto(double valor) {costo=valor;}
	
	public double getCosto() {return costo;}


	public enum Tipo{
		
		CAMISA(25000),PANTALON(30000),ABRIGO(27000);
		
		private double precioEstandar;
		
		private Tipo(double precioEstandar) {
			this.setPrecioEstandar(precioEstandar);
		}

		public double getPrecioEstandar() {
			return precioEstandar;
		}

		public void setPrecioEstandar(double precioEstandar) {
			this.precioEstandar = precioEstandar;
		}
		
		
		
	}

	@Override
	public String toString() {
		return "\nTipo: "+nombre+"\nPrecio: "+precio;
	}
	
	public Tipo getTipo() {
		return tipo;
	}

	public void setTipo(Tipo tipo) {
		this.tipo = tipo;
	}
	
	//Metodos ENVIO
		public static ArrayList<Producto> seleccionarProductos(SETS set,int cantidad) {
			ArrayList<Producto> productos=new ArrayList<Producto>();
			switch(set) {
			case ONLY: 
				for(int i=0;i<cantidad;i++) {
					productos.add(new Producto(Tipo.CAMISA));
				
			}
			break;
			case TU: 
				for(int i=0;i<cantidad;i++) {
					productos.add(new Producto(Tipo.CAMISA));
					productos.add(new Producto(Tipo.PANTALON));
				
			}
			break;
			case COMPLETO: 
				for(int i=0;i<cantidad;i++) {
					productos.add(new Producto(Tipo.CAMISA));
					productos.add(new Producto(Tipo.PANTALON));
					productos.add(new Producto(Tipo.ABRIGO));
					
				
			}
			break;
			
			}
			return productos;
		}
	    
		public static ArrayList<Producto> clasificar(ArrayList<Producto> productos, Tipo tipo){
			ArrayList<Producto> clasificados=new ArrayList<Producto>();
			for(int i=0;i<productos.size();i++) {
				Tipo tipado=productos.get(i).getTipo();
				if(tipado==tipo) {
					clasificados.add(productos.get(i));
				}
			}
			return clasificados;
		}


		
	
	
}

File: Python/src/gestorAplicacion/clasesBase/Producto.py
import pickle

class SETS(Enum):
	ONLY =1
	TU = 2
	COMPLETO = 3


class Producto:
    class Tipo:
        CAMISA = 40000
        PANTALON = 50000
        ABRIGO = 60000

        def __init__(self, precioEstandar):
            self.precioEstandar = precioEstandar

    def __init__(self, tipo, precio=None, costo=None):
        self.nombre = tipo.name
        self.precio = tipo.precioEstandar if precio is None else precio
        self.costo = costo


    # def __eq__(self, p):
    #     if isintansce(p, Producto):
    #         if self.nombre==p.getNombre():
    #             return True
    #         else:
    #             return False






    def getNombre(self)->str:
        return self.nombre

    def setNombre(self, nombre):
        self.nombre=nombre

    def getPrecio(self)->float:
        return self.precio

    def setPrecio(self, precio:float):
        self.precio=precio

    def getCosto(self)->float:
        return self.costo

    def setCosto(self, costo:float):
        self.costo=costo

    def __str__(self):
        return "\nTipo: "+self.nombre+"\nPrecio: "+self.precio;


	@staticmethod
	def seleccionar_productos(set, cantidad):
		productos = []
		if set == SETS.ONLY:
			for _ in range(cantidad):
				productos.append(Producto(Producto.Tipo.CAMISA))
		elif set == SETS.TU:
			for _ in range(cantidad):
				productos.append(Producto(Producto.Tipo.CAMISA))
				productos.append(Producto(Producto.Tipo.PANTALON))
		elif set == SETS.COMPLETO:
			for _ in range(cantidad):
				productos.append(Producto(Producto.Tipo.CAMISA))
				productos.append(Producto(Producto.Tipo.PANTALON))
				productos.append(Producto(Producto.Tipo.ABRIGO))
		return productos

	@staticmethod
	def clasificar(productos, tipo):
		clasificados = []
		for producto in productos:
			if producto.tipo == tipo:
				clasificados.append(producto)
		return clasificados